### PR TITLE
Reuse the array key of mimetypes

### PR DESCRIPTION
**Pull request by LukasReschke**
This function is called a lot of times and was really slow before due to not reusing the same array.

Previously when it was called 500'000 times it took about 2seconds, now we're down to 0.2 seconds on my local machine.

Ref https://github.com/owncloud/core/issues/13434

### CODE REVIEW

**Comment by Xenopathic** [comment]
Metrics? But I can see why this is important, so :+1:

**Comment by LukasReschke** [comment]
Well, there is not much sense in a metric that shows exactly the same as my original comment says ;-)

**Comment by MorrisJobke** [comment]
fpalm30c3

I will test this.

**Comment by MorrisJobke** [comment]
I uploaded my sample files collection that consists of many different file types. All mimetypes are fine there. So :+1: from my side.

@DeepDiver1975 @karlitschek Low change of breaking stuff, but huge performance impact (as it even can be cached in PHPs OPCache ;)) I would rate for a merge into master before ownCloud 8

**Comment by karlitschek** [comment]
looks good :+1:

File: lib/private/helper.php [hunk 1]
```diff
@@ -28,6 +28,75 @@ class OC_Helper {
 	private static $mimetypeIcons = array();
 	private static $mimetypeDetector;
 	private static $templateManager;
+	/** @var string[] */
+	private static $mimeTypeAlias = array(
```
Comment (DeepDiver1975) [inline lib/private/helper.php]: I'd really prefer to get this out of the helper class - but this is something for 8.1

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**